flocker/control/_protocol.py

```diff
@@ -107,10 +111,101 @@ def version(self):
 
     @NodeStateCommand.responder
     def node_changed(self, hostname, node_state):
-        self.cluster_state.update_node_state(hostname, node_state)
+        self.control_amp_service.node_changed(hostname, node_state)
         return {}
 
 
+class ControlAMP(AMP):
+    """
+    AMP protocol for control service server.
+    """
+    def __init__(self, control_amp_service):
+        """
+        :param ControlAMPService control_amp_service: The service managing AMP
+             connections to the control service.
+        """
+        AMP.__init__(self, locator=ControlServiceLocator(control_amp_service))
+        self.control_amp_service = control_amp_service
+
+    def connectionMade(self):
+        AMP.connectionMade(self)
+        self.control_amp_service.connected(self)
+
+    def connectionLost(self, reason):
+        AMP.connectionLost(self, reason)
+        self.control_amp_service.disconnected(self)
+
+
+class ControlAMPService(Service):
+    """
+    Control Service AMP server.
+
+    Convergence agents connect to this server.
+    """
+    def __init__(self, cluster_state, configuration_service, endpoint):
+        """
+        :param ClusterStateService cluster_state: Object that records known
+            cluster state.
+        :param ConfigurationPersistenceService configuration_service:
+            Persistence service for desired cluster configuration.
+        :param endpoint: Endpoint to listen on.
+        """
+        self.connections = set()
+        self.cluster_state = cluster_state
+        self.configuration_service = configuration_service
+        self.endpoint_service = StreamServerEndpointService(
+            endpoint, ServerFactory.forProtocol(lambda: ControlAMP(self)))
+        # When configuration changes, notify all connected clients:
+        self.configuration_service.register(
+            lambda: self._send_state_to_connections(self.connections))
```

Comment:
Yeah, that was note to myself; I finished it up later before putting it up for review.